### ISSUE TITLE
[Solution2] currently broken

### ISSUE THREAD

**Issue by palladius** [palladius/clouddeploy-platinum-path#34]
Solution 2 is currently broken.

Looks like Canary works for both Python and Ruby, but  Prod has zero selectors.

![AjuCi5ZqDTBC5Yo](https://user-images.githubusercontent.com/132802/201367621-432e3934-f053-42fc-8336-e1869a80de12.png)

**Comment by palladius** [comment]
Note app01 does have 4 pods in prod, they're just not selected correctly.

**Comment by palladius** [comment]
Since CANARY was working and PROD was not selecting it, it was relatively easy to find:

I'm basically reversing a part of this CL: 

https://github.com/palladius/clouhttps://github.com/palladius/clouddeploy-platinum-path/commit/dd345338bcf052ed9fdafce284d8c02ea7f8e3f4

**Comment by palladius** [comment]
Since Daniel fixed manually my project032 - i'll try the new code for project031.
Code executed, but nothing changes. I believe I need for a fresh app01/02 to make it to prod :) 

    $ bin/bump-versions all
    $ git commit -a -m 'bumping...'

Now promoting DEV to PROD as all best SREs in the world :P

**Comment by palladius** [comment]
pushed the first via UI, the second from CLI cos showing off is what I do:

```
(base) ricc@derek2:🏡~/git/clouddeploy-platinum-path$ ./10-auto-promote-APP_XX-STAGE_YY-to-STAGE_ZZ.sh app02 dev production
========================================
== Script START: './10-auto-promote-APP_XX-STAGE_YY-to-STAGE_ZZ.sh' (on 20221111-162023)
========================================
Now promoting DEV to STAG for PIPELINE=app02 (from ARGV1) and RELEASE=app02-20221111-1512-v2-1-5autobump..
+ gcloud deploy releases promote --to-target production --region us-central1 --release app02-20221111-1512-v2-1-5autobump --delivery-pipeline=app02 --quiet
```

**Comment by palladius** [comment]
Woops, computer says no. While DEV worked find, PROD was broken (note to self, being able to test CANA/PROD before getting to the stage).

```
"ERROR: build step 0 "us-central1-docker.pkg.dev/cd-image-prod/cd-image/cd@sha256:8b4bd730cf7ec608340cfe3d33ac672ce9709f5b09c856c724c0fbca9b53cecc" failed: step exited with non-zero status: 1"
```

Reverting completely the code, and re-promoting once it makes to dev for just app01 until Im happy. in 7d1567b552a0bbfe2fd6d0b0ab61e999db9ab8b7

**Comment by palladius** [comment]
.. and it's now fixed as of v1.0.4 as of 26b72a39a9d2c0299d45c13b25988c2600ebdc44

**Comment by palladius** [comment]
```
🐼 bin/solution2-simple-curl app02
[DEBUG] IP_FWDRULE[app01]: 34.117.168.161
app=app01 version=2.35autobump target=prod emoji=🐍
app=app01 version=2.35autobump target=prod emoji=🐍
app=app01 version=2.35autobump target=prod emoji=🐍
app=app01 version=2.35autobump target=prod emoji=🐍
app=app01 version=2.35autobump target=prod emoji=🐍
app=app01 version=2.35autobump target=prod emoji=🐍
app=app01 version=2.35autobump target=prod emoji=🐍
app=app01 version=2.35autobump target=canary emoji=🐍
app=app01 version=2.35autobump target=prod emoji=🐍
app=app01 version=2.35autobump target=prod emoji=🐍
app=app01 version=2.35autobump target=canary emoji=🐍
app=app01 version=2.35autobump target=canary emoji=🐍
[DEBUG] IP_FWDRULE[app02]: 34.117.241.206
app=app02 emoji=💎 target=prod version=2.1.7autobump
app=app02 emoji=💎 target=prod version=2.1.7autobump
app=app02 emoji=💎 target=prod version=2.1.7autobump
app=app02 emoji=💎 target=prod version=2.1.7autobump
app=app02 emoji=💎 target=prod version=2.1.7autobump
app=app02 emoji=💎 target=canary version=2.1.7autobump
app=app02 emoji=💎 target=prod version=2.1.7autobump
app=app02 emoji=💎 target=prod version=2.1.7autobump
app=app02 emoji=💎 target=canary version=2.1.7autobump
app=app02 emoji=💎 target=prod version=2.1.7autobump
app=app02 emoji=💎 target=prod version=2.1.7autobump
app=app02 emoji=💎 target=prod version=2.1.7autobump
```
